src/lib/ContractManager.js

```diff
@@ -151,19 +152,27 @@ export default class ContractManager {
     return contract;
   }
 
-  async getErc20Data(address) {
-    const contract = new ethers.Contract(address, erc20Abi, this.getEthersProvider());
+  async getTokenData(address, type) {
+    const contract = new ethers.Contract(address, type === 'erc20' ? erc20Abi : erc721Abi, this.getEthersProvider());
     try {
-      const decimals = await contract.decimals();
-      if (!decimals)
+      let tokenData = {};
+      tokenData.name = await contract.name.call();
+      if (!tokenData.name)
         return;
 
-      const name = await contract.name.call();
-      const symbol = await contract.symbol.call();
-      if (!symbol || !name)
+      tokenData.symbol = await contract.symbol.call();
+
+      if (type == 'erc20') {
+        tokenData.decimals = await contract.decimals();
+      } else if (type == 'erc721') {
+        // can't be sure if this contract would support ERC721Metadata
+        //tokenData.baseURI = await contract.baseURI();
+      }
+
+      if (!tokenData.symbol || !tokenData.name)
```

Comment:
don't need the second part of or conditional `!tokenData.name` is returned on line 160 so it would always have it here